src/DateComparison.php

```diff
@@ -34,7 +34,7 @@ public static function isTomorrow(DateTimeImmutable $date): bool
 
     public static function isCurrentWeek(DateTimeImmutable $date): bool
     {
-        return CarbonImmutable::instance($date)->isCurrentWeek();
+        return self::inTheFuture($date) && CarbonImmutable::instance($date)->isCurrentWeek();
```

Comment:
I'm a bit conflicted by this check being here inside `isCurrentWeek`. Because if we later want to use `isCurrentWeek` for another scenario where it does not matter if the day is in the past or not, this will probably result in some confusion. Also, "isCurrentWeek" is not influenced by the day being in the past or future.

Maybe we can extract it to a method `isUpcomingDayInCurrentWeek` that combines `self::inTheFuture()` and `self::isCurrentWeek()`

And small nit: Maybe `isCurrentWeek()` should be `inCurrentWeek()`? Or `inTheFuture` should be `isInTheFuture()`?

Reply:
Aside from that the change look good functionally! 👍

Reply:
Good idea, I've made the changes.

Reply:
Thanks! I meant renaming one of the two though. Now every method starts with "is" except for "inCurrentWeek()" it seems. So for consistency that can remain "isCurrentWeek", or alternatively it could be "isInCurrentWeek()"?

Reply:
There is still `onSameDay()` maybe I should rename that one as well to `isOnTheSameDay()` , maybe I should change that as well, then all functions are in the same `is...()`?

Reply:
We can also merge it like this, it's super minor